qiskit_experiments/calibration_management/backend_calibrations.py

```diff
@@ -97,13 +115,23 @@ def __init__(
 
             # Add the basis gates
             for gate in library.basis_gates:
-                self.add_schedule(library[gate])
+                self.add_schedule(library[gate], n_qubits=library.n_qubits(gate))
 
             # Add the default values
             for param_conf in library.default_values():
                 schedule_name = param_conf[-1]
                 if schedule_name in library.basis_gates:
-                    self.add_parameter_value(*param_conf)
+                    # Call super to avoid inst_map update as not all parameters
+                    # will have default values until this loop completes.
+                    super().add_parameter_value(*param_conf)
+
+        # Push the schedules to the instruction schedule map.
+        self.update_inst_map()
+
+    @property
+    def instruction_schedule_map(self) -> InstructionScheduleMap:
+        """Returns the updated instruction schedule map."""
+        return self._inst_map
```

Comment:
Does it actually take much time to generate a full `InstructionScheduleMap`? I wonder if keeping it constantly updated is worth the effort versus just generating it on demand.

Reply:
I have not benchmarked it. I would assume that the more schedules and qubits one has to manage the more expensive it gets. I see a few options for the inst. map
1. Current implementation: update inst. map on parameter update.
2. Generate whole inst. map each time the inst. map is required.
3. Generate a partial inst. map based on the qubits needed.

Option 2. will definitely do lots of unnecessary steps which might be slow for setups with many qubits. I cannot say how much this overhead will be as I haven't benchmarked it but to be on the safe side I avoided Option 2. I think Option 3. would make requesting an inst. map less trivial as one would need to specify the qubits we work with. For now I would thus stick with option 1.

Reply:
Yes, the current implementation seems fine. It might be interesting to benchmark calibration handling code in general though I don't think performance is a high priority right now. It would be interesting to know how long it takes to generate a 100 qubit InstructionScheduleMap for instance.

I think there might be other spots in calibrations.py that could be faster if they had an extra look up table instead of looping over keys looking for specific values but we should wait until we really study performance.

Reply:
This constantly updated InstructionScheduleMap might highlight an issue with the `Calibrations` API that we should think about. In `Calibrations`, `get_schedule` and `get_parameter_value` accept parameters `cutoff_date`, `group`, and `valid_only` to filter which parameters get used but in this PR `BackendCalibrations.add_parameter_value` always updates InstructionScheduleMap. A few questions this raises for me:

1. What if one wants an `InstructionScheduleMap` filtered by a group or date?
2. How do we expect users to specify group and parameter validity?
3. Should we adjust the way the `Calibrations` API works with parameter history? Perhaps separate the live values and the history a bit? Maybe `Calibrations` should just keep the latest values like the `InstructionScheduleMap` in this PR, and then there could be a `load_from_db` type of method that could use a cutoff date and other filters to update the current parameter values in the `Calibrations` instance from values in the database.

Reply:
Good points. Commit 308634a should address point 1. Now, if the user wants an `inst_map` from a different group then `update_inst_map` can be called with the desired group and cutoff date to regenerate the inst_map. The other two points seem like a discussion that transcends this PR.